compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1220,6 +1211,9 @@ convertExpr env0 e = do
             pure $ ETmLam (v, TStruct fields) $ ERecCon tupleType $ zipWithFrom mkFieldProj (1 :: Int) fields
     go env (VarIn GHC_Types "primitive") (LType (isStrLitTy -> Just y) : LType t : args)
         = fmap (, args) $ convertPrim (envLfVersion env) (unpackFS y) <$> convertType env t
+    -- erase mkMethod calls and leave only the body.
+    go env (VarIn DA_Internal_Desugar "mkMethod") (LType _tpl : LType _iface : LType _methodName : LType _methodTy : LExpr _implDict : LExpr _hasMethodDic : LExpr body : args)
```

Comment:
I feel slightly uneasy about the fact that we’re changing the type here. This is fine since we never have call sites to a method defined via `mkMethod` but it seems like something that might blow up at some point. Don’t really have a better suggestion though so seems sensible to roll with this for now.

Reply:
It should be mostly fine anyway since `Method t i m r` is just a newtype for `t -> r`, right? 

If it does become a problem in the future, I think we could add an argument to `convertExpr` for custom "conversion passes". Then, we could move this `go` clause to a separate "conversion pass", and only use it from `convertImplements`, while the general `convertExpr` would be equivalent to the new `convertExpr` with `id` (or `pure`, or similar) as the "conversion pass".

Reply:
Unfortunately newtypes in daml become record types in LF, so it's not coercible. OTOH we have an LF typechecker so if anyone tries to call `mkMethod` outside of erased code, it should be a type error.

Reply:
I see! thanks for explaining that @sofiafaro-da. I think we can feel comfortable given the second part of your comment :)

Reply:
I added a test just in case! https://github.com/digital-asset/daml/pull/12175